pytorch_lightning/trainer/supporters.py

```diff
@@ -119,27 +122,249 @@ def mean(self):
         return self.total / self.num_values
 
 
+class LightningBatchSamplerWrapper:
+    """
+    This class wraps user batch sampler, so we can extract
+    the batch_indices for tracking each sample.
+    """
+
+    def __init__(self, batch_sampler):
+        self.batch_sampler = batch_sampler
+        self.batch_indices = None
+
+    def __iter__(self):
+        for batch_indices in self.batch_sampler:
+            self.batch_indices = batch_indices
+            yield batch_indices
+
+    @staticmethod
+    def to_new_dataloader(dataloader: DataLoader) -> DataLoader:
+        """
+        This function will wrap the user batch_sampler to track the returned batch indices
+        """
+        if not isinstance(dataloader, DataLoader):
+            raise MisconfigurationException('Autoid only works with torch dataloaders or derived classes!')
+
+        if isinstance(dataloader.dataset, IterableDataset):
+            return dataloader
+
+        skip_keys = ['sampler', 'batch_sampler', 'dataset_kind']
+        skip_valid_keys = ['args', 'kwargs', 'self']
+
+        params = {k:v for k, v in vars(dataloader).items() if not k.startswith("_")}
+
+        valid_kwargs = set(inspect.signature(dataloader.__init__).parameters)
+        contains_dataset = True
+
+        if type(dataloader) is not DataLoader:
```

Comment:
```suggestion
        if not isinstance(dataloader, DataLoader):
```
if do I miss something...